File: app/src/main/java/com/example/mike/mp3player/client/views/SeekerBar.java
package com.example.mike.mp3player.client.views;

import android.animation.ValueAnimator;
import android.content.Context;
import android.util.AttributeSet;
import android.view.View;

import androidx.appcompat.widget.AppCompatSeekBar;

import com.example.mike.mp3player.R;

public class SeekerBar extends AppCompatSeekBar {

    private ValueAnimator valueAnimator;
    private boolean isTracking = false;

    public SeekerBar(Context context) {
        this(context, null);
    }

    public SeekerBar(Context context, AttributeSet attrs) {
        this(context, attrs, R.attr.seekBarStyle);
    }

    public SeekerBar(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        this.setVisibility(VISIBLE);
        this.setLayerType(View.LAYER_TYPE_SOFTWARE, null);
    }

    @Override
    public final void setOnSeekBarChangeListener(OnSeekBarChangeListener listener) {
        super.setOnSeekBarChangeListener(listener);
    }

    public ValueAnimator getValueAnimator() {
        return valueAnimator;
    }

    public void setValueAnimator(ValueAnimator valueAnimator) {
        this.valueAnimator = valueAnimator;
    }

    public boolean isTracking() {
        return isTracking;
    }

    public void setTracking(boolean tracking) {
        isTracking = tracking;
    }

}

File: app/src/main/java/com/github/goldy1992/mp3player/client/views/SeekerBar.kt
package com.github.goldy1992.mp3player.client.views

import android.animation.ValueAnimator
import android.content.Context
import android.util.AttributeSet
import android.view.View
import androidx.appcompat.widget.AppCompatSeekBar
import com.github.goldy1992.mp3player.R

class SeekerBar
@JvmOverloads constructor(context: Context?,
                          attrs: AttributeSet? = null,
                          defStyleAttr: Int = R.attr.seekBarStyle)
    : AppCompatSeekBar(context, attrs, defStyleAttr) {

    var valueAnimator: ValueAnimator? = null
    var isTracking = false
    override fun setOnSeekBarChangeListener(listener: OnSeekBarChangeListener) {
        super.setOnSeekBarChangeListener(listener)
    }

    init {
        this.visibility = View.VISIBLE
        setLayerType(View.LAYER_TYPE_SOFTWARE, null)
    }
}